homeassistant/components/tesla/__init__.py

```diff
@@ -134,7 +134,16 @@ async def async_setup_entry(hass, config_entry):
     """Set up Tesla as config entry."""
     hass.data.setdefault(DOMAIN, {})
     config = config_entry.data
-    websession = aiohttp_client.async_get_clientsession(hass)
+    count = 0
+    if hass.config_entries.async_entries(DOMAIN):
+        for entry in hass.config_entries.async_entries(DOMAIN):
+            if entry.title == config_entry.title:
+                break
+            count += 1
+    if count > 0:
+        websession = aiohttp_client.async_create_clientsession(hass)
```

Comment:
IMHO. Just always create a new session. This just make it more complex to little benefit. But might want to wait for somebody else on the team to chime in.

Reply:
I agree. Create away.

Reply:
What about the quality guidance to share the session? This is the only place we would have the complexity.

Reply:
One reason for the use of ha session is for shared pool. This i agree we loose here. We still get the benefit of common user agents by creating a new object using hass functions.

An alternative solution here would be to use a custom cookie jar and pass along to the request. Or possibly by sharing the TcpConnector instance instead (though i don't think that we have investigated that yet).

I just feel the extra complexity outweighs the benefit of the shared session here.

Reply:
I'm not sure the 7 or so lines of code to determine if a config entry exists is that complex but I can just create the session.

Reply:
Make sure to add a comment explaining for future readers why we're having different sessions.

Reply:
Ohh. just for reference. I just checked code async_create_clientsession().. it actually does re-use connection pool so it's perfectly fine to use ;)

Reply:
Changes done.